biz.aQute.resolve/test/biz/aQute/resolve/RunResolutionTest.java

```diff
@@ -517,24 +518,40 @@ public void testPrintHumanReadableDifference() throws Exception {
 	}
 
 	@Test
-	public void testStartLevelDecoration() throws Exception {
-		Bndrun bndrun = Bndrun.createBndrun(workspace, IO.getFile(ws.toFile(), "test.simple/resolve.bndrun"));
+	public void testStartLevelDecoration(SoftAssertions softly) throws Exception {
+		Bndrun bndrun = Bndrun.createBndrun(workspace, IO.getFile(ws.toFile(), "test.simple/resolveduplicates.bndrun"));
 		bndrun.setProperty("-runstartlevel", "order=leastdependenciesfirst,begin=100,step=10");
 
 		// Decorate test.simple to get startlevel 90 (which would otherwise be 110 within the assigned runstartlevel).
 		bndrun.setProperty("-runbundles+", "test.simple;startlevel=90");
 
 		List<? extends HeaderClause> runBundles = List.copyOf(bndrun.resolve(false, false, new NoopConverter<>()));
 
-		assertThat(runBundles).hasSize(2);
-		assertThat(runBundles.get(0)
-			.getName()).isEqualTo("osgi.enroute.junit.wrapper");
-		assertThat(runBundles.get(0)
-			.getAttribs()).containsEntry(Constants.RUNBUNDLES_STARTLEVEL_ATTRIBUTE, "100");
-		assertThat(runBundles.get(1)
+		softly.assertThat(runBundles).hasSize(4);
+		softly.assertThat(runBundles.get(0)
+			.toString()).isEqualTo("org.apache.felix.gogo.runtime;version='[0.12.0,0.12.1)';startlevel=100");
+		softly.assertThat(runBundles.get(1)
+			.toString()).isEqualTo("org.apache.felix.gogo.runtime;version='[0.10.0,0.10.1)';startlevel=110");
+		softly.assertThat(runBundles.get(2)
+			.toString()).isEqualTo("osgi.enroute.junit.wrapper;version='[4.12.0,4.12.1)';startlevel=120");
+
+		softly.assertThat(runBundles.get(3)
 			.getName()).isEqualTo("test.simple");
-		assertThat(runBundles.get(1)
+		softly.assertThat(runBundles.get(3)
 			.getAttribs()).containsEntry(Constants.RUNBUNDLES_STARTLEVEL_ATTRIBUTE, "90");
+		softly.assertThat(runBundles.get(3)
+			.toString()).isEqualTo("test.simple;version=snapshot;startlevel=90");
```

Comment:
This assertion is almost a duplicate of the one on line 540/541.

Reply:
fixed by recent commit.